fem/transfer.cpp

```diff
@@ -288,10 +294,333 @@ void L2ProjectionGridTransfer::L2Projection::ElemMixedMass(
    }
 }
 
-L2ProjectionGridTransfer::L2ProjectionL2Space::L2ProjectionL2Space(
-   const FiniteElementSpace &fes_ho_, const FiniteElementSpace &fes_lor_)
-   : L2Projection(fes_ho_, fes_lor_)
+void L2ProjectionGridTransfer::L2Projection::ElemMixedMass(
+   Geometry::Type geom, const FiniteElement& fe_ho,
+   const FiniteElement& fe_lor, ElementTransformation* el_tr,
+   IntegrationPointTransformation& ip_tr,
+   DenseMatrix& B_L, DenseMatrix& B_H) const
+{
+   int order = fe_lor.GetOrder() + fe_ho.GetOrder() + el_tr->OrderW();
+   const IntegrationRule* ir = &IntRules.Get(geom, order);
+
+   DenseMatrix M_mixed_el(fe_lor.GetDof(), fe_ho.GetDof());
+   M_mixed_el = 0.0;
+
+   for (int i = 0; i < ir->GetNPoints(); i++)
+   {
+      const IntegrationPoint& ip_lor = ir->IntPoint(i);
+      IntegrationPoint ip_ho;
+
+      //maps integration point ip_lor -> ip_ho
+      ip_tr.Transform(ip_lor, ip_ho);
+      Vector shape_lor(fe_lor.GetDof());
+      fe_lor.CalcShape(ip_lor, shape_lor);
+      Vector shape_ho(fe_ho.GetDof());
+      fe_ho.CalcShape(ip_ho, shape_ho);
+
+      for (int j=0; j<shape_lor.Size(); ++j)
+      {
+         B_L(i, j) = shape_lor(j);
+      }
+
+      for (int j=0; j<shape_ho.Size(); ++j)
+      {
+         B_H(i, j) = shape_ho(j);
+      }
+   }
+
+}
+
+Vector L2ProjectionGridTransfer::L2Projection::MixedMassEA(
+   const FiniteElementSpace& fes_ho_ea,
+   const FiniteElementSpace& fes_lor_ea,
+   const Coefficient *coeff_, Vector &M_LH, MemoryType d_mt_)
+{
+   Mesh* mesh_ho = fes_ho_ea.GetMesh();
+   Mesh* mesh_lor = fes_lor_ea.GetMesh();
+   int nel_ho = mesh_ho->GetNE();
+   int nel_lor = mesh_lor->GetNE();
+   int ndof_ho = fes_ho_ea.GetNDofs();
+   int ndof_lor = fes_lor_ea.GetNDofs();
+
+   QuadratureFunction qfunc;
+   IntegrationRule ir;
+   if (coeff_ == nullptr)
+   {
+      Geometry::Type geom = mesh_ho->GetElementBaseGeometry(0);
+      const FiniteElement &fe = *fes_ho_ea.GetFE(0);
+      const FiniteElement &fe_lor = *fes_lor_ea.GetFE(0);
+      ElementTransformation *el_tr = fes_lor_ea.GetElementTransformation(0);
+      int qorder = fe_lor.GetOrder() + fe.GetOrder() + el_tr->OrderW();
+      ir = IntRules.Get(geom, qorder);
+
+      QuadratureSpace qspace(*mesh_lor, ir);
+      qfunc = QuadratureFunction(&qspace);
+      qfunc = 1.0;
+   }
+   else
+   {
+      // dynamic_cast to check if QuadFuncCoeff; if yes, continue; if no, return error
+      auto qfunc_coeff = dynamic_cast<const QuadratureFunctionCoefficient*>(coeff_);
+      if (qfunc_coeff == NULL)
+      {
+         mfem_error("Not a QuadratureFunctionCoefficient - for inside H1");
+      }
```

Comment:
Instead of `mfem_error`, use `MFEM_VERIFY` because it will add some context info to the error message -- function name plus file and line number:
```suggestion
      MFEM_VERIFY(qfunc_coeff != NULL,
                  "Only QuadratureFunctionCoefficient is currently supported");
```